src/Simplify.elm

```diff
@@ -9052,6 +9060,90 @@ resultToMaybeCompositionChecks checkInfo =
         ()
 
 
+resultFromMaybeChecks : CheckInfo -> Maybe (Error {})
+resultFromMaybeChecks checkInfo =
+    fromMaybeChecks
+        { onNothingFn = ( [ "Result" ], "Err" ), onJustFn = ( [ "Result" ], "Ok" ) }
+        checkInfo
+
+
+fromMaybeChecks : { onNothingFn : ( ModuleName, String ), onJustFn : ( ModuleName, String ) } -> CheckInfo -> Maybe (Error {})
+fromMaybeChecks config checkInfo =
```

Comment:
Why the split into 2 functions? Do you intend to reuse `fromMaybeChecks` on another type?

Reply:
So we can reuse it for e.g.
- [Json.Decode.Extra.fromMaybe](https://package.elm-lang.org/packages/elm-community/json-extra/4.3.0/Json-Decode-Extra#fromMaybe)
- [RemoteData.fromMaybe](https://package.elm-lang.org/packages/krisajenkins/remotedata/6.0.1/RemoteData#fromMaybe)
- [Yaml.Decode.fromMaybe](https://package.elm-lang.org/packages/MaybeJustJames/yaml/latest/Yaml-Decode#fromMaybe)
- [Csv.Decode.fromMaybe](https://package.elm-lang.org/packages/BrianHicks/elm-csv/latest/Csv-Decode#fromMaybe)

Reply:
Alright, sure.